### DISCUSSION TITLE
wazuh agent trying to connect to security onion , but SO refusing connection

### DISCUSSION THREAD

**Discussion by brebe** [Security-Onion-Solutions/securityonion#5709] [Unsupported Versions] [answered]
hello dear guys,
i am trying to enroll a linux agent to the SO wazuh system.
i did 
1. so-allow > w >and enter ip 
2. sudo so-wazuh-agent-manage > extracted the key
3. entered in the conf the key and ip and restarted the agent

the logs saying that the agent is trying  to connect to server but server is refusing the connection 
what am i doing wrong ? thanks

**Comment by Sahale** [comment]
Hi! Did you configure firewall how explained in docs?

**Comment by brebe** [comment]
yes, 
i forgot to mention i have the CENTOS image installed.

**Comment by dougburks** [comment]
Are there any network firewall or other devices between the agent and the manager which may be interfering with the traffic?

**Comment by LS3-Allen** [comment]
Try this command:
sudo so-firewall includehost wazuh_agent <ip address>

**Comment by brebe** [comment]
thanks for all your answers,
there is no firewall between the SO and the machine with the wazuh agent ( which i am trying to monitor with the wazuh agent)
ive been studying all day so the situation i have is : 
1.  my SO is standalone / airgap
2. sudo so-firewall includehost wazuh_agent 192.168.0.1
3. i even edited manually  assigned_hostgroups.map.yaml and added - {{ portgroups.wazuh_agent }} to the standalone role INPUT chain.
4. i dont understand how to add this into the /opt/so/saltstack/local/pillar/minions/securityonion_standalone.sls (in my case) what is the sintax for this beacuse i tried like : 
```
firewall:
  assigned_hostgroups:
    chain:
      DOCKER-USER:
        hostgroups:
          wazuh_agent:
            portgroups:
             - portgroups.wazuh_agent
```
and in the end i did  it ssudo salt securityonion_standalone state.apply firewall
but it still doesnt work

**Reply by LS3-Allen** [reply]
If the steps above didn't work then it could be a network connection issue between the Wazuh agent and the SO server.  Can you ping the SO server from the Wazuh agent(192.168.0.1)?

**Reply by brebe** [reply]
yes i can ping and i can ssh from agent machine to SO

**Reply by LS3-Allen** [reply]
Is the agent configured to connect on UDP port 1514 or TCP port 1514?

**Reply by brebe** [reply]
right now tcp but at a moment i remember i switched to udp

**Reply by LS3-Allen** [reply]
Try to telnet from the agent machine to the SO server using TCP 1514.  If that does not work then you definitely have a network blockage between the agent and SO.  If telnet does work then it would be a misconfiguration between the agent and SO.

**Reply by brebe** [reply]
hehe, i already test  comm to  opened port, i will try to reinstall SO as eval, not as standalone i believe is something with setting in pillars minions of salt

**Comment by NetwarSystem** [comment, marked as answer]
Perhaps it would be helpful to do this:

ssh to SO machine

tshark -i <local network device> host <thing that can not connect>

and you can do similar stuff on the client side. Just verify you have a working path between the two machines, then you can start going up the stack in terms of config stuff.

**Reply by brebe** [reply]
thank you all for answers,
 i solved the problem, i dont know why and what it was the problem, but, i opened the port 1514 tcp on centos firewall clasically (firewall-cmd) and after that i created another group called wazuh-agent and enrolled the agent into this group and it started to work, i am receiving from agent messages. i dont know waht was the problem basically but i solved it, thank you guys for advices.

**Reply by DinhDiesel** [reply]
@brebe do you know what exactly you did to resolve this? created a group "wazuh-agent" and enrolled the agent into this group fixed this?

**Comment by FritzLott** [comment]
I am (hopelessly?) confused about firewalld.  Since Wazuh is working for you, would you please post a listing of public.xml (cat /usr/lib/firewalld/zones/public.xml)?  Thank you!

I would like to know what I should be aiming towards!
